### PR TITLE
Add base-url option

### LINKED ISSUE #428: Make an API call to a self-hosted/enterprise instance
**Is your feature request related to a problem? Please describe.**
Users would like to make API calls to a self-hosted instance of GitHub Enterprise Server using this action. Currently that is not possible by setting GITHUB_API_URL, as this environment variable is not honored. To demonstrate this issue, use the following workflow...

```
name: Test

on:
  workflow_dispatch:

jobs:
  test:
    name: Test
    runs-on: ubuntu-latest

    env:
      GHES_API_URL: https://my.github-enterprise-server.com/api/v3

    steps:
      - name: Demonstrate GITHUB_API_URL
        uses: actions/github-script@v6
        env:
          GITHUB_API_URL: ${{ env.GHES_API_URL }}
        with:
          github-token: xxxxx
          script: |
            console.log(`Demonstrate the resolution of GITHUB_API_URL environment variable...`);
            console.log(`GITHUB_API_URL: ${process.env.GITHUB_API_URL}`);
```

The log of this workflow shows that the `GITHUB_API_URL` is set to `https://api.github.com/` despite being explicitly set otherwise in the workflow. Here is the output...

```
Run actions/github-script@v6
  with:
    github-token: xxxxx
    script: console.log(`Demonstrate the resolution of GITHUB_API_URL environment variable...`);
  console.log(`GITHUB_API_URL: ${process.env.GITHUB_API_URL}`);
  
    debug: false
    user-agent: actions/github-script
    result-encoding: json
    retries: 0
    retry-exempt-status-codes: 400,401,403,404,4[2](https://github.com/robandpdx/junk/actions/runs/6589941226/job/17905485324#step:2:2)2
  env:
    GHES_API_URL: https://my.github-enterprise-server.com/api/v[3](https://github.com/robandpdx/junk/actions/runs/6589941226/job/17905485324#step:2:3)
    GITHUB_API_URL: https://my.github-enterprise-server.com/api/v3
Demonstrate the resolution of GITHUB_API_URL environment variable...
GITHUB_API_URL: https://api.github.com/
```

This was previously discussed in #150 and in #151. However the proposed solution in #151 does not work, as demonstrated above.

**Describe the solution you'd like**
Adding an optional BASE_URL as proposed in #151 would solve this issue.

**Describe alternatives you've considered**
I have forked this repo and added the needed option. However, others would benefit from this change as well

**Additional context**
Nothing to see here.

### PR DESCRIPTION
**Pull request by robandpdx**
Resolves #428

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -29,6 +29,9 @@ inputs:
   retry-exempt-status-codes:
     description: A comma separated list of status codes that will NOT be retried e.g. "400,500". No effect unless `retries` is set
     default: 400,401,403,404,422 # from https://github.com/octokit/plugin-retry.js/blob/9a2443746c350b3beedec35cf26e197ea318a261/src/index.ts#L14
+  base-url:
+    description: The root of the API URL of the GHES instance to which to connect.
```
Comment (joshmgross) [inline action.yml:33]: What do you think about calling this the "GitHub REST API" URL to match the terminology we use to describe the `github.api_url` in our docs?

https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

File: src/main.ts
Comment (joshmgross) [inline src/main.ts:1]: You'll need to run `npm run build` to generate the `dist/` changes - https://github.com/actions/github-script/blob/1f16022c7518aad314c43abcd029895291be0f52/package.json#L10

File: action.yml (hunk 1 above)
Comment (robandpdx) [reply action.yml:33]: Sure!

Comment (joshmgross) [reply action.yml:33]: Maybe something like:
```suggestion
    description: An optional GitHub REST API to connect to a different GitHub instance.
```

Comment (joshmgross) [reply action.yml:33]: And maybe include an example like `https://my.github-enterprise-server.com/api/v3` (is that the right form?)

**Review by joshmgross — APPROVED** [review]

File: action.yml (hunk 1 above)
Comment (robandpdx) [reply action.yml]: Done!

**Comment by gonmmarques** [comment]
Can this be merged? I'm also facing what was explained here https://github.com/actions/github-script/issues/428

**Comment by gustavkj** [comment]
Ironically, this broke workflows on Github Enterprise. I think the `undefined` value still will be interpreted as the value being set.

It worked before with any extra config, since Github Enterprise set the `GITHUB_API_URL` environment variable (just like Github.com, [see docs](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)) and it used by the `@actions/github` package to set the base url here: https://github.com/actions/toolkit/blob/20f826bfe76164099ab2403d9ea8509e16843223/packages/github/src/internal/utils.ts#L42-L44

One solution would be to set a default value to be the pre-configured value, or fix the assignment to the `opts` object only when it is set.

I've opened a bug: https://github.com/actions/github-script/issues/436

**Pull request merged**